### PR TITLE
Add workflow to run tests

### LINKED ISSUE #6: Run Django tests in GitHub Actions

### PR DESCRIPTION
**Pull request by selamanse**
Closes #6 

I added a separate workflow to run the tests on PRs and pushes to master.
Looking at the circleci configuration I understood that you were using poetry with python 3.7 to manage the project, so I adopted this to the workflow and all seems to run nicely.

Please have a look and also make suggestions if you need anything else.

### CODE REVIEW

File: .github/workflows/run-tests.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        python-version: ${{ matrix.python-version }}
+    - name: Set up Poetry ${{ matrix.poetry-version }}
+      uses: abatilo/actions-poetry@v2.0.0
+      with:
+        poetry-version: ${{ matrix.poetry-version }}
+    - name: Install dependencies
+      working-directory: ./api
+      run: |
+        poetry install
+    - name: Style Guide Enforcement
```
Comment (zkan) [inline .github/workflows/run-tests.yml:35]:
```suggestion
    - name: Style guide enforcement
```

File: .github/workflows/run-tests.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      working-directory: ./api
+      run: |
+        poetry run flake8 testroad
+    - name: Run Tests
```
Comment (zkan) [inline .github/workflows/run-tests.yml:39]:
```suggestion
    - name: Run tests
```

**Review by zkan — CHANGES_REQUESTED** [review]
@selamanse Thanks! I'd request some changes, so that the text style looks consistent.

**Review by selamanse — COMMENTED** [review]
👍

**Comment by selamanse** [comment]
@zkan I cleaned up a little by adding defaults (reducing the duplicate declaration of workspace) and removing comments from the template I used.

**Review by zkan — APPROVED** [review]
Looks great! Thanks. :)

**Pull request merged**